src/kbmod/search/raw_image_eigen.h

```diff
@@ -0,0 +1,250 @@
+    // and returning them if they fit. That'd work well in combination with centered_block
+    std::array<unsigned, 4> centered_block(const unsigned size, const unsigned width, const unsigned height){
+      unsigned d = size/2;
+      unsigned left_x = ((i-d >= 0) && (i-d < width)) ? i-d : i;
+      unsigned right_x = ((i+d >= 0) && (i+d < width)) ? i+d : width - i;
+      unsigned top_y = ((j-d >= 0) && (j-d < height)) ? j-d : j;
+      unsigned bot_y = ((j+d >= 0) && (j+d < height)) ? j+d : height - i;
+
+      unsigned dx = right_x - left_x;
+      unsigned dy = bot_y - top_y;
+      return {left_x, dx, top_y, dy};
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, const Index& rc);
+  };
+
+  std::ostream& operator<<(std::ostream& os, const Index& rc){
+    os << "x: " << rc.i << " y: " << rc.j;
+    return os;
+  }
+
+
+
+  struct Point{
+    float x;
+    float y;
+
+    Point(float xd, float yd)
+      : x(xd), y(yd) {}
+
+    Index to_index(){
+      return Index(x, y);
+    }
+
+    std::array<Point, 4> nearest_pixel_coords()  {
+      return {
+        Point(floor(x-0.5f)+0.5f, floor(y+0.5f)+0.5f),
+        Point(floor(x+0.5f)+0.5f, floor(y+0.5f)+0.5f),
+        Point(floor(x-0.5f)+0.5f, floor(y-0.5f)+0.5f),
+        Point(floor(x+0.5f)+0.5f, floor(y-0.5f)+0.5f)
+      };
+    }
+
+    std::array<Index, 4> nearest_pixel_idxs(){
+      return {
+        Index(x-0.5f, y+0.5f),
+        Index(x+0.5f, y+0.5f),
+        Index(x-0.5f, y-0.5f),
+        Index(x+0.5f, y-0.5f)
+      };
+    };
+
+    friend std::ostream& operator<<(std::ostream& os, const Point& rc);
+  };
+
+  std::ostream& operator<<(std::ostream& os, const Point& rc){
+    os << "x: " << rc.x << " y: " << rc.y;
+    return os;
+  }
+
+
+  // templates brother
+  using Image = Eigen::Matrix<float, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+  using ImageI = Eigen::Matrix<int, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+  using ImageRef = Eigen::Ref<Image>;
+  using ImageIRef = Eigen::Ref<Image>;
+
+
+  class RawImageEigen {
+  public:
+    RawImageEigen();
+    explicit RawImageEigen(Image& img, double obs_time=-1.0d);
+    explicit RawImageEigen(unsigned w, unsigned h, float value=0.0f, double obs_time=-1.0d);
+
+    RawImageEigen(const RawImageEigen& old);  // Copy constructor
+    RawImageEigen(RawImageEigen&& source);    // Move constructor
+
+    RawImageEigen& operator=(const RawImageEigen& source);  // Copy assignment
+    RawImageEigen& operator=(RawImageEigen&& source);       // Move assignment
+
+    // Basic getter functions for image data.
+    unsigned get_width() const { return width; }
+    unsigned get_height() const { return height; }
+    unsigned get_npixels() const { return width * height; }
+
+    // Inline pixel functions.
+    float get_pixel(int x, int y) const {
+      return Index(x, y).is_within(width, height) ? image(x, y) : NO_DATA;
```

Comment:
It does; and it doesn't.

This doesn't matter because the method has been left in so I can run a copy of `test_raw_image.py` with a `import kbmod.RawImageEigen as RawImage`. I fudged around trying to write some supporting infrastructure to help me interact with the matrix from Python, but it's hopeless Like I said last Monday, I don't know of a way to expose getters and setters of `Image` so that we can in-place edit from Python as if it was a `numpy.array`. I should have been explicit about this in the Draft PR text box at the start and save you the effort.

Removed it.